src/global.rs

```diff
@@ -19,15 +27,41 @@ use sync::list::List;
 lazy_static! {
     /// REGISTRIES is the head pointer of the list of mutator registries.
     pub static ref REGISTRIES: List<LocalEpoch> = List::new();
+    /// GARBAGES is a reference to the global queue of garbages.
+    pub static ref GARBAGES: Queue<(usize, Bag)> = Queue::new();
     /// EPOCH is a reference to the global epoch.
     pub static ref EPOCH: Epoch = Epoch::new();
 }
 
 
+#[inline]
+pub fn push_bag<'scope>(bag: &mut Bag, scope: &'scope Scope) {
+    let epoch = EPOCH.load(Relaxed);
+    let bag = ::std::mem::replace(bag, Bag::new());
+    ::std::sync::atomic::fence(SeqCst);
+    GARBAGES.push((epoch, bag), scope);
+}
+
 /// Collect several bags from the global old garbage queue and destroys their objects.
+///
 /// Note: This may itself produce garbage and in turn allocate new bags.
 pub fn collect(scope: &Scope) {
-    unimplemented!()
+    let epoch = EPOCH.try_advance(&REGISTRIES, scope);
+
+    let condition = |bag: &(usize, Bag)| {
+        // A pinned thread can witness at most one epoch advancement. Therefore, any bag that is
+        // within one epoch of the current one cannot be destroyed yet.
+        let diff = epoch.wrapping_sub(bag.0);
+        cmp::min(diff, 0usize.wrapping_sub(diff)) > 2
+    };
+
+    let garbages = &GARBAGES;
+    for _ in 0..COLLECT_STEPS {
+        match garbages.try_pop_if(&condition, scope) {
```

Comment:
I just recalled that @stjepang commented on it: https://github.com/crossbeam-rs/crossbeam-epoch/pull/3#issuecomment-318157956

In short, it's wrong..  I will read how `coco` dealt with it, and think how to resolve this issue here.